docs/src/man/basics.md

```diff
@@ -0,0 +1,1417 @@
+ 67
+ 22
+ 49
+ 45
+ 53
+ 35
+ 53
+ 35
+ 61
+ 28
+  ⋮
+ 34
+ 23
+ 30
+ 50
+ 31
+ 40
+ 38
+ 23
+ 27
+```
+
+By default `select` copies columns of a passed source data frame. In order to avoid copying, pass `copycols=false`:
+
+```jldoctest dataframe
+julia> df = select(german, :Sex)
+1000×1 DataFrame
+  Row │ Sex
+      │ String
+──────┼────────
+    1 │ male
+    2 │ female
+    3 │ male
+    4 │ male
+    5 │ male
+    6 │ male
+    7 │ male
+    8 │ male
+  ⋮   │   ⋮
+  994 │ male
+  995 │ male
+  996 │ female
+  997 │ male
+  998 │ male
+  999 │ male
+ 1000 │ male
+985 rows omitted
+
+julia> df.Sex === german.Sex
+false
+
+julia> df = select(german, :Sex, copycols=false)
+1000×1 DataFrame
+  Row │ Sex
+      │ String
+──────┼────────
+    1 │ male
+    2 │ female
+    3 │ male
+    4 │ male
+    5 │ male
+    6 │ male
+    7 │ male
+    8 │ male
+  ⋮   │   ⋮
+  994 │ male
+  995 │ male
+  996 │ female
+  997 │ male
+  998 │ male
+  999 │ male
+ 1000 │ male
+985 rows omitted
+
+julia> df.Sex === german.Sex
+true
+```
+
+To perform the selection operation in-place use `select!`:
+
+```jldoctest dataframe
+julia> german = copy(german_ref);
+
+julia> select!(german, Not(:Age));
+
+julia> german
+1000×9 DataFrame
+  Row │ id     Sex     Job    Housing  Saving accounts  Checking account  Cred ⋯
+      │ Int64  String  Int64  String   String           String            Int6 ⋯
+──────┼─────────────────────────────────────────────────────────────────────────
+    1 │     0  male        2  own      NA               little                 ⋯
+    2 │     1  female      2  own      little           moderate
+    3 │     2  male        1  own      little           NA
+    4 │     3  male        2  free     little           little
+    5 │     4  male        2  free     little           little                 ⋯
+    6 │     5  male        1  free     NA               NA
+    7 │     6  male        2  own      quite rich       NA
+    8 │     7  male        3  rent     little           moderate
+  ⋮   │   ⋮      ⋮       ⋮       ⋮            ⋮                ⋮               ⋱
+  994 │   993  male        3  own      little           little                 ⋯
+  995 │   994  male        2  own      NA               NA
+  996 │   995  female      1  own      little           NA
+  997 │   996  male        3  own      little           little
+  998 │   997  male        2  own      little           NA                     ⋯
+  999 │   998  male        2  free     little           little
+ 1000 │   999  male        2  own      moderate         moderate
+                                                  3 columns and 985 rows omitted
+```
+
+`transform` and `transform!` functions work identically to `select` and `select!` with the only difference that
+they retain all columns that are present in the source data frame and another difference is that `transform` and
+`transform!` always copy columns when column renaming transformation is passed.
```

Comment:
maybe give an example?

Reply:
Yeah, I have discussed it in a different section of another file.

Reply:
I do not see this in this PR. Where do you discuss it?

Reply:
Not in this PR I just put some text/note. In my next PR, I have covered. The order of content won't change. I have noted in my points that all content is in order or not when I will mention these files in `make.jl`.

Reply:
As commented several times - it is very hard to review PRs if they are not self contained. I think we should not merge PRs that rely on other PRs that are not reviewed nor merged yet. @nalimilan - could you please comment what you think? Thank you!

Reply:
It would make sense to have an example of `transform` here indeed. Any reason not to include it in this PR?